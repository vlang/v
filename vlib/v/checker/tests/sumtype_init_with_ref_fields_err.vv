struct Node {
	parent &Node
	left   &Node
	right  &Node
}

type Sumtype = Node | int

fn main() {
	s := Sumtype{}
	dump(s)
}
