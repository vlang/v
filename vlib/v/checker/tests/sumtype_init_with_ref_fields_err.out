vlib/v/checker/tests/sumtype_init_with_ref_fields_err.vv:10:7: error: reference field `Node.parent` must be initialized
    8 | 
    9 | fn main() {
   10 |     s := Sumtype{}
      |          ~~~~~~~~~
   11 |     dump(s)
   12 | }
vlib/v/checker/tests/sumtype_init_with_ref_fields_err.vv:10:7: error: reference field `Node.left` must be initialized
    8 | 
    9 | fn main() {
   10 |     s := Sumtype{}
      |          ~~~~~~~~~
   11 |     dump(s)
   12 | }
vlib/v/checker/tests/sumtype_init_with_ref_fields_err.vv:10:7: error: reference field `Node.right` must be initialized
    8 | 
    9 | fn main() {
   10 |     s := Sumtype{}
      |          ~~~~~~~~~
   11 |     dump(s)
   12 | }
